core/src/main/scala/io/finch/metrics/Metrics.scala

```diff
@@ -0,0 +1,53 @@
+package io.finch.metrics
+
+trait Metrics {
+
+  /**
+    * Count a call to endpoint by path and method
+    */
+  def countCall(path: String): Unit
+
+  /**
+    * Count failure if exception was thrown during processing request to an endpoint
+    */
+  def countFailure(path: String, e: Throwable): Unit
+
+  /**
+    * Count time that was required to process failed request
+    */
+  def failureTime(path: String, e: Throwable, duration: Float): Unit
+
+  /**
+    * Count success if request was processed without exceptions
+    */
+  def countSuccess(path: String, code: Int): Unit
+
+  /**
+    * Count
+    */
+  def successTime(path: String, code: Int, duration: Float): Unit
+
+  /**
+    * Count size of response
+    */
+  def size(path: String, s: Float): Unit
```

Comment:
Nit: Why just `s` and not `size`? Also, why count the size of messages seperately from the responses?

Reply:
Sorry, I don't get the question about size of message

Reply:
I was wondering why the size was counted through a seperate method instead of combining it like with `successTime` or `successCount`. Now there is now way to correlate the size of responses with the status code or response time.

Reply:
Ah, I see.

It makes sense